src/components/actions/evaluatorFunctions.ts

```diff
@@ -8,6 +8,42 @@ Any changes done here should also be replicated in front-end
 
 import { DateTime, Duration } from 'luxon'
 
+type FilterRule = 'exclude' | 'include'
+interface FilterOptions {
+  key?: string
+  rule?: FilterRule
+  values: string | string[]
+}
+
+const filterArray = (valuesArray: Object[] | string[], options: FilterOptions) => {
+  const { key, rule = 'exclude', values } = options
+  if (
+    (valuesArray.length > 0 && key && typeof valuesArray[0] === 'object') ||
+    (!key && typeof valuesArray[0] === 'string')
+  )
+    return key
+      ? (valuesArray as Object[]).filter(
+          (
+            obj // Array of strings
+          ) =>
+            Object.entries(obj).find(
+              ([keyMap, value]) =>
+                keyMap === key &&
+                (rule === 'include' ? values.includes(value) : !values.includes(value))
+            )
+        )
+      : (valuesArray as string[]).filter(
+          (
+            str // Array of objects
+          ) => (rule === 'include' ? values.includes(str) : !values.includes(str))
```

Comment:
It's nice `values.includes` works for both arrays and strings, but unfortunately it doesn't work for other primitive types.

Probably best to convert it all to an array at the top:
```
const compareValues = Array.isArray(values) ? values : [ values ]
```